app/src/components/PlaybackControl.tsx

```diff
@@ -0,0 +1,266 @@
+                handlePlayback(true);
+                this.getPlaybackOrderBookData();
+            }
+        }
+    };
+
+    /**
+     * Checks the real time rate to see whether it is valid or not.
+     * @returns {boolean}
+     */
+    checkRealTimeRate = (): boolean => {
+        let valid: boolean = true;
+        const realTimeRate = this.getRealTimeRate();
+        if (realTimeRate > MAXIMUM_PLAYBACK_REAL_TIME_RATE) {
+            this.showMessage(`Please select a lower real time rate for playback. Maximum ratio is 5. `
+             + `You entered: ${realTimeRate}`);
+            this.setState({ unitSpeed: MAXIMUM_PLAYBACK_REAL_TIME_RATE });
+            valid = false;
+        }
+        return valid;
+    };
+
+    /**
+     * @desc Gets the real time rate for playback, 0 if Messages is selected as unit
+     * @returns {number}
+     */
+    getRealTimeRate = (): number => {
+        const { unitSpeed, selectedUnit } = this.state;
+        if (selectedUnit === 'Messages') return 0;
+
+        let timeInNano: number;
+        switch (selectedUnit) {
+        case 'Seconds':
+            timeInNano = NANOSECONDS_IN_ONE_SECOND;
+            break;
+        case 'Milliseconds':
+            timeInNano = NANOSECONDS_IN_ONE_MILLISECOND;
+            break;
+        case 'Microseconds':
+            timeInNano = NANOSECONDS_IN_ONE_MICROSECOND;
+            break;
+        case 'Nanoseconds':
+            timeInNano = 1;
+            break;
+        default:
+            timeInNano = NANOSECONDS_IN_ONE_SECOND;
+        }
+        return (timeInNano / NANOSECONDS_IN_ONE_SECOND) * unitSpeed;
+    };
+
+    /**
+     * @desc Gets playback speed parameter based on selected unit and unit speed
+     */
+    getPlaybackParameter = (): string => {
+        const { unitSpeed, selectedUnit } = this.state;
+        let parameter: string = '?delay=2.5&';
+        if (selectedUnit === 'Messages') parameter = `${parameter}rateMessages=${unitSpeed}`;
+        else {
+            parameter = `${parameter}rateRealtime=${this.getRealTimeRate()}`;
+        }
+        return parameter;
+    };
+
+    /**
+     * @desc Calls the backend service to get new data to feed OrderBook, graph
+     */
+    getPlaybackOrderBookData = (): void => {
+        const {
+            selectedInstrument, lastSodOffset, handlePlaybackModifications,
+        } = this.props;
+        const parameter = this.getPlaybackParameter();
+
+        const endPoint = `ws://localhost:5050/playback/${selectedInstrument}/${lastSodOffset}/${parameter}`;
+
+        this.playbackWS = new WebSocket(endPoint);
+        this.playbackWS.onopen = () => {
+            console.log('opened playback websocket');
+        };
+        this.playbackWS.onmessage = m => {
+            const data: PlaybackData = JSON.parse(m.data);
+            handlePlaybackModifications(data);
+        };
+        this.playbackWS.onclose = () => {
+            console.log('closed playback websocket');
+        };
+    };
```

Comment:
Should we put this code in some service file?

Reply:
Im not too sure, cause normal calls just make a request through the service and then handle the data with some processing and set states

Reply:
I still think the handlers should be in the service file. Especially since you are not using onclose/onopen, then you would only need to pass in the onmessage handler as a parameter

Reply:
But then what do you suggest for closing the web socket ? we put it as a static variable ?

Reply:
Currently there is axios logging for HTTP requests, do we want to remove the console logs for opening and closing the socket?

Reply:
Perhaps you can store it as a class variable like this guy: https://dev.to/finallynero/using-websockets-in-react-4fkp

Reply:
thsts what i did initially. it was a class var in playback

Reply:
now its staitc in orderbook service